x-pack/plugins/security_solution/public/timelines/components/timeline/unified_components/data_table/index.tsx

```diff
@@ -0,0 +1,495 @@
+        );
+      },
+      [dispatch, timelineId]
+    );
+
+    const onResizeDataGrid = useCallback(
+      (colSettings) => {
+        onColumnResize({ columnId: colSettings.columnId, width: Math.round(colSettings.width) });
+      },
+      [onColumnResize]
+    );
+
+    const onAddFilter = useCallback(
+      (field: DataViewField | string, values: unknown, operation: '+' | '-') => {
+        if (dataView && filterManager) {
+          const fieldName = typeof field === 'string' ? field : field.name;
+          popularizeField(dataView, fieldName, dataViews, capabilities);
+          const newFilters = generateFilters(filterManager, field, values, operation, dataView);
+          return filterManager.addFilters(newFilters);
+        }
+      },
+      [filterManager, dataView, dataViews, capabilities]
+    );
+
+    const onChangeItemsPerPage = useCallback(
+      (itemsChangedPerPage) =>
+        dispatch(
+          timelineActions.updateItemsPerPage({ id: timelineId, itemsPerPage: itemsChangedPerPage })
+        ),
+      [dispatch, timelineId]
+    );
+
+    const browserFieldsByName = useMemo(() => getAllFieldsByName(browserFields), [browserFields]);
+
+    const customColumnRenderers = useMemo(
+      () =>
+        getFormattedFields({
+          dataTableRows: tableRows,
+          scopeId: 'timeline',
+          headers: columns,
+          browserFieldsByName,
+        }),
+      [browserFieldsByName, columns, tableRows]
+    );
+
+    const handleFetchMoreRecords = useCallback(() => {
+      onChangePage(fetchedPage + 1);
+      setFechedPage(fetchedPage + 1);
+    }, [fetchedPage, onChangePage]);
+
+    const additionalControls = useMemo(
+      () => <ToolbarAdditionalControls timelineId={timelineId} updatedAt={updatedAt} />,
+      [timelineId, updatedAt]
+    );
+
+    const cellActionsMetadata = useMemo(() => ({ scopeId: timelineId }), [timelineId]);
+
+    const onUpdateSampleSize = useCallback(
+      (newSampleSize: number) => {
+        if (newSampleSize !== sampleSize) {
+          dispatch(timelineActions.updateSampleSize({ id: timelineId, sampleSize: newSampleSize }));
+          refetch();
+        }
+      },
+      [dispatch, sampleSize, timelineId, refetch]
+    );
+
+    const onUpdateRowHeight = useCallback(
+      (newRowHeight: number) => {
+        if (newRowHeight !== rowHeight) {
+          dispatch(timelineActions.updateRowHeight({ id: timelineId, rowHeight: newRowHeight }));
+        }
+      },
+      [dispatch, rowHeight, timelineId]
+    );
+
+    const services = useMemo(() => {
+      return {
+        theme,
+        fieldFormats,
+        storage,
+        toastNotifications: toastsService,
+        uiSettings,
+        dataViewFieldEditor,
+        data: dataPluginContract,
+      };
+    }, [
+      theme,
+      fieldFormats,
+      storage,
+      toastsService,
+      uiSettings,
+      dataViewFieldEditor,
+      dataPluginContract,
+    ]);
+
+    if (!dataView) {
+      return null;
```

Comment:
Please pull in an error view or something here. Maybe work with docs for an error like "Table cannot be displayed. DataView not detected". Probably unlikely to happen, but just in case...

Reply:
This is done here: 642109f0dd2